packages/smooth_app/lib/pages/prices/prices_proofs_page.dart

```diff
@@ -58,149 +139,182 @@ class _PricesProofsPageState extends State<PricesProofsPage>
           ),
         ],
       ),
-      body: FutureBuilder<MaybeError<GetProofsResult>>(
-        future: _results,
-        builder: (
-          final BuildContext context,
-          final AsyncSnapshot<MaybeError<GetProofsResult>> snapshot,
-        ) {
-          if (snapshot.connectionState != ConnectionState.done) {
-            return const Center(child: CircularProgressIndicator());
-          }
-          if (snapshot.hasError) {
-            return Text(snapshot.error!.toString());
-          }
-          // highly improbable
-          if (!snapshot.hasData) {
-            return const Text('no data');
-          }
-          if (snapshot.data!.isError) {
-            return Text(snapshot.data!.error!);
-          }
-          final GetProofsResult result = snapshot.data!.value;
-          // highly improbable
-          if (result.items == null) {
-            return const Text('empty list');
-          }
-          final double squareSize = MediaQuery.sizeOf(context).width / _columns;
-
-          final AppLocalizations appLocalizations =
-              AppLocalizations.of(context);
-          final String title = result.numberOfPages == 1
-              ? appLocalizations.prices_proofs_list_length_one_page(
-                  result.items!.length,
-                )
-              : appLocalizations.prices_proofs_list_length_many_pages(
-                  _pageSize,
-                  result.total!,
-                );
-          return Column(
-            children: <Widget>[
-              SmoothCard(
-                child: ListTile(
-                  title: Text(title),
-                ),
+      body: _bearerToken == null
+          ? // Show loading while authenticating
+          const Center(child: CircularProgressIndicator())
+          : // Show content once authenticated
+          _buildProofsContent(context, appLocalizations),
+    );
+  }
+
+  Widget _buildProofsContent(
+    BuildContext context,
+    AppLocalizations appLocalizations,
+  ) {
+    return Column(
+      children: <Widget>[
+        Builder(
```

Comment:
I don't get the point of the `Builder` layer.
Please comment it or get rid of it.